src/probability/kernel/integral_comp_prod.lean

```diff
@@ -0,0 +1,344 @@
+/-
+Copyright (c) 2023 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import probability.kernel.composition
+
+/-!
+# Bochner integral of a function against the composition-product of two kernels
+
+We prove properties of the composition-product of two kernels. If `κ` is an s-finite kernel from
+`α` to `β` and `η` is an s-finite kernel from `α × β` to `γ`, we can form their composition-product
+`κ ⊗ₖ η : kernel α (β × γ)`. We proved in `probability.kernel.lintegral_comp_prod` that it verifies
+`∫⁻ bc, f bc ∂((κ ⊗ₖ η) a) = ∫⁻ b, ∫⁻ c, f (b, c) ∂(η (a, b)) ∂(κ a)`. In this file, we prove the
+same equality for the Bochner integral.
+
+## Main statements
+
+* `probability_theory.integral_comp_prod`: the integral against the composition-product is
+  `∫ z, f z ∂((κ ⊗ₖ η) a) = ∫ x, ∫ y, f (x, y) ∂(η (a, x)) ∂(κ a)`
+
+## Implementation details
+
+This file is to a large extent a copy of part of `measure_theory.constructions.prod`. The product of
+two measures is a particular case of composition-product of kernels and it turns out that once the
+measurablity of the Lebesgue integral of a kernel is proved, almost all proofs about integral
+against products of measures extend with minimal modifications to the composition-product of two
+kernels.
+-/
+
+noncomputable theory
+open_locale topology ennreal measure_theory probability_theory
+open set function real ennreal measure_theory filter probability_theory
+
+variables {α β γ E : Type*}
+  {mα : measurable_space α} {mβ : measurable_space β} {mγ : measurable_space γ}
+  [normed_add_comm_group E]
+  {κ : kernel α β} [is_s_finite_kernel κ]
+  {η : kernel (α × β) γ} [is_s_finite_kernel η]
+  {a : α}
+
+namespace probability_theory
+
+lemma ae_kernel_lt_top (a : α) {s : set (β × γ)} (hs : measurable_set s) (h2s : (κ ⊗ₖ η) a s ≠ ∞) :
```

Comment:
You shouldn't need `hs` here, as `s` is included in another set which is measurable and of the same measure (see `measure_theory.to_measurable`).